### PR TITLE
[Transform] add retention policy to documentation IT

### PR DESCRIPTION
**Pull request by hendrikmuhs**
add retention policy and sync to documentation IT, add builders for sync
and retention policy

relates #67832

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeRetentionPolicyConfig.java [hunk 1]
```diff
@@ -96,4 +96,37 @@ public int hashCode() {
...
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String field;
+        private TimeValue maxAge;
+
+        /**
+         * The time field that defines the creation time of a document.
```
Comment (davidkyle) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeRetentionPolicyConfig.java:109]:
```suggestion
         * The time field used to calculate the age of a document.
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeRetentionPolicyConfig.java [hunk 2]
```diff
@@ -96,4 +96,37 @@ public int hashCode() {
...
+         * @return The {@link Builder} with the field set.
+         */
+        public Builder setField(String field) {
+            this.field = field;
+            return this;
+        }
+
+        /**
+         * The max age, all documents that are older will be deleted.
+         * @param maxAge
```
Comment (przemekwitek) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeRetentionPolicyConfig.java:120]: Missing description

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeSyncConfig.java [hunk 3]
```diff
@@ -94,4 +96,37 @@ public String getName() {
...
+         */
+        public Builder setField(String field) {
+            this.field = field;
+            return this;
+        }
+
+        /**
+         * The time delay between the current time and the latest input data time.
+         * The default value is 60s.
+         * @param delay
```
Comment (przemekwitek) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TimeSyncConfig.java:120]: Missing description

**Review by davidkyle — APPROVED** [review]
LGTM

I was looking for the docs changes to put_transform.asciidoc but I guess that is in #68768

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**